apps/service_providers/llm_service/adapters.py

```diff
@@ -155,17 +98,33 @@ def for_experiment(cls, experiment: Experiment, session: ExperimentSession, trac
             temperature=experiment.temperature,
             prompt_text=experiment.prompt_text,
             max_token_limit=experiment.max_token_limit,
-            experiment_version_number=experiment.version_number,
-            experiment_is_a_version=experiment.is_a_version,
             tools=get_tools(session, experiment=experiment),
             input_formatter=experiment.input_formatter,
             source_material_id=experiment.source_material_id,
             trace_service=trace_service,
         )
 
-    @staticmethod
-    def for_pipeline(experiment: Experiment, session: ExperimentSession) -> Self:
-        """TODO"""
+    @classmethod
+    def for_pipeline(
+        cls,
+        session: ExperimentSession,
+        node: "AssistantNode",
+        llm_service: LlmService,
+        provider_model: "LlmProviderModel",
+    ) -> Self:
+        return cls(
+            session=session,
+            provider_model_name=provider_model.name,
+            llm_service=llm_service,
+            temperature=node.llm_temperature,
+            prompt_text=node.prompt,
+            max_token_limit=provider_model.max_token_limit,
+            tools=get_tool_instances(node.tools, session),
+            input_formatter="{input}",
+            source_material_id=node.source_material_id,
+            trace_service=session.experiment.trace_service,  # TODO
```

Comment:
Is this TODO still relevant?

Reply:
Ah excellent, my todo's are doing its job (I put that there incase I forgot to bring it up and someone were to catch them in a review. I should do a better job and reviewing my own PRs...).

This one was for me to address the absence of trace provider configuration for pipelines. I'm thinking it would make sense to configure them at the experiment level. If you agree, I can quickly add that change to this PR (would be super small)

Reply:
I'll address the todos in a new PR